patches/components-embedder_support-user_agent_utils.cc.patch

```diff
@@ -10,3 +10,12 @@ index c957a691efcac5c4574fae032a876d97ad60fafc..196bb8cdf26b0f5395a33e520d63bf90
    absl::optional<std::string> maybe_brand_override =
        base::GetFieldTrialParamValueByFeature(features::kGreaseUACH,
                                               "brand_override");
+@@ -492,6 +493,8 @@ blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+ 
+ #if BUILDFLAG(IS_WIN)
+   metadata.platform_version = GetWindowsPlatformVersion();
++#elif BUILDFLAG(IS_LINUX)
++  metadata.platform_version = "";
```

Comment:
Thanks for the review. Unfortunately, I believe such trick with macros is not viable in some environments. In particular, I get the following error:
```
> autoninja -C /home/anton/Desktop/brave/pc/src/out/Component brave -k 1 --offline
ninja: Entering directory `/home/anton/Desktop/brave/pc/src/out/Component'
[1/5] CXX obj/components/embedder_support/browser_util/user_agent_utils.o
In file included from ../../brave/chromium_src/components/embedder_support/user_agent_utils.cc:24:
../../../src/components/embedder_support/user_agent_utils.cc:442:10: error: no matching function for call to 'GetUserAgentMetadata_ChromiumImpl'
  442 |   return GetUserAgentMetadata(nullptr, only_low_entropy_ch);
      |          ^~~~~~~~~~~~~~~~~~~~
../../brave/chromium_src/components/embedder_support/user_agent_utils.cc:22:30: note: expanded from macro 'GetUserAgentMetadata'
   22 | #define GetUserAgentMetadata GetUserAgentMetadata_ChromiumImpl
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../src/components/embedder_support/user_agent_utils.cc:441:26: note: candidate function not viable: requires single argument 'only_low_entropy_ch', but 2 arguments were provided
  441 | blink::UserAgentMetadata GetUserAgentMetadata(bool only_low_entropy_ch) {
      |                          ^                    ~~~~~~~~~~~~~~~~~~~~~~~~
../../brave/chromium_src/components/embedder_support/user_agent_utils.cc:22:30: note: expanded from macro 'GetUserAgentMetadata'
   22 | #define GetUserAgentMetadata GetUserAgentMetadata_ChromiumImpl
      |                              ^
```

This is because the first implementation `GetUserAgentMetadata` refers to the second one. Specifically, the first implementation takes one argument gets parsed prior to the second definition which takes two, and at the time it is parsed and token is substituted, the second one is not substituted yet. If I change the order of functions (cut the first one and paste it after the second one), the error goes away. I'll paste the diff below, but I do not think it is a viable solution.

The 3-line patch I posted earlier is likely the cleanest solution.

Reply:
```diff
diff --git a/chromium_src/components/embedder_support/user_agent_utils.cc b/chromium_src/components/embedder_support/user_agent_utils.cc
index 5681ef02c1..0ebe5ffed0 100644
--- a/chromium_src/components/embedder_support/user_agent_utils.cc
+++ b/chromium_src/components/embedder_support/user_agent_utils.cc
@@ -19,6 +19,21 @@ constexpr char kBraveBrandNameForCHUA[] = "Brave";
 // constructing the UA in brave/browser/brave_content_browser_client.cc, but we
 // can't use it here in the //components.
 #define BRAVE_GET_USER_AGENT_BRAND_LIST brand = kBraveBrandNameForCHUA;
+#define GetUserAgentMetadata GetUserAgentMetadata_ChromiumImpl
 
 #include "src/components/embedder_support/user_agent_utils.cc"
 #undef BRAVE_GET_USER_AGENT_BRAND_LIST
+#undef GetUserAgentMetadata
+
+namespace embedder_support {
+
+blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+                                              bool only_low_entropy_ch) {
+  blink::UserAgentMetadata metadata = GetUserAgentMetadata_ChromiumImpl(pref_service, only_low_entropy_ch);
+#if BUILDFLAG(IS_LINUX)
+  metadata.platform_version = "";
+#endif
+  return metadata;
+}
+
+}
```

Reply:
Diff to be applied to `brave-core`:
```diff
diff --git a/chromium_src/components/embedder_support/user_agent_utils.cc b/chromium_src/components/embedder_support/user_agent_utils.cc
index 5681ef02c1..0ebe5ffed0 100644
--- a/chromium_src/components/embedder_support/user_agent_utils.cc
+++ b/chromium_src/components/embedder_support/user_agent_utils.cc
@@ -19,6 +19,21 @@ constexpr char kBraveBrandNameForCHUA[] = "Brave";
 // constructing the UA in brave/browser/brave_content_browser_client.cc, but we
 // can't use it here in the //components.
 #define BRAVE_GET_USER_AGENT_BRAND_LIST brand = kBraveBrandNameForCHUA;
+#define GetUserAgentMetadata GetUserAgentMetadata_ChromiumImpl
 
 #include "src/components/embedder_support/user_agent_utils.cc"
 #undef BRAVE_GET_USER_AGENT_BRAND_LIST
+#undef GetUserAgentMetadata
+
+namespace embedder_support {
+
+blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+                                              bool only_low_entropy_ch) {
+  blink::UserAgentMetadata metadata = GetUserAgentMetadata_ChromiumImpl(pref_service, only_low_entropy_ch);
+#if BUILDFLAG(IS_LINUX)
+  metadata.platform_version = "";
+#endif
+  return metadata;
+}
+
+}
diff --git a/patches/components-embedder_support-user_agent_utils.cc.patch b/patches/components-embedder_support-user_agent_utils.cc.patch
index 241e474687..697708e4b9 100644
--- a/patches/components-embedder_support-user_agent_utils.cc.patch
+++ b/patches/components-embedder_support-user_agent_utils.cc.patch
@@ -10,3 +10,26 @@ index c957a691efcac5c4574fae032a876d97ad60fafc..196bb8cdf26b0f5395a33e520d63bf90
    absl::optional<std::string> maybe_brand_override =
        base::GetFieldTrialParamValueByFeature(features::kGreaseUACH,
                                               "brand_override");
+@@ -437,9 +438,11 @@ std::string GetPlatformForUAMetadata() {
+ #endif
+ }
+ 
++/*
+ blink::UserAgentMetadata GetUserAgentMetadata(bool only_low_entropy_ch) {
+   return GetUserAgentMetadata(nullptr, only_low_entropy_ch);
+ }
++*/
+ 
+ blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+                                               bool only_low_entropy_ch) {
+@@ -505,6 +508,10 @@ blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+   return metadata;
+ }
+ 
++blink::UserAgentMetadata GetUserAgentMetadata(bool only_low_entropy_ch) {
++  return GetUserAgentMetadata(nullptr, only_low_entropy_ch);
++}
++
+ #if BUILDFLAG(IS_ANDROID)
+ void SetDesktopUserAgentOverride(content::WebContents* web_contents,
+                                  const blink::UserAgentMetadata& metadata,
```

Reply:
I believe you just need to forward-declare `GetUserAgentMetadata_ChromiumImpl` version manually in the `.cc`.

```diff
diff --git a/chromium_src/components/embedder_support/user_agent_utils.cc b/chromium_src/components/embedder_support/user_agent_utils.cc
index 5681ef02c1..a92e6de642 100644
--- a/chromium_src/components/embedder_support/user_agent_utils.cc
+++ b/chromium_src/components/embedder_support/user_agent_utils.cc
@@ -11,6 +11,14 @@ constexpr char kBraveBrandNameForCHUA[] = "Brave";

 }  // namespace

+namespace embedder_support {
+
+blink::UserAgentMetadata GetUserAgentMetadata_ChromiumImpl(
+    const PrefService* local_state,
+    bool only_low_entropy_ch = false);
+
+}  // namespace embedder_support
+
 // Chromium uses `version_info::GetProductName()` to get the browser's "brand"
 // name, but on MacOS we use different names for different channels (adding Beta
 // or Nightly, for example). In the UA client hint, though, we want a consistent
@@ -19,6 +27,22 @@ constexpr char kBraveBrandNameForCHUA[] = "Brave";
 // constructing the UA in brave/browser/brave_content_browser_client.cc, but we
 // can't use it here in the //components.
 #define BRAVE_GET_USER_AGENT_BRAND_LIST brand = kBraveBrandNameForCHUA;
+#define GetUserAgentMetadata GetUserAgentMetadata_ChromiumImpl

 #include "src/components/embedder_support/user_agent_utils.cc"
 #undef BRAVE_GET_USER_AGENT_BRAND_LIST
+#undef GetUserAgentMetadata
+
+namespace embedder_support {
+
+blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service,
+                                              bool only_low_entropy_ch) {
+  blink::UserAgentMetadata metadata =
+      GetUserAgentMetadata_ChromiumImpl(pref_service, only_low_entropy_ch);
+#if BUILDFLAG(IS_LINUX)
+  metadata.platform_version = "";
+#endif
+  return metadata;
+}
+
+}  // namespace embedder_support
```

Reply:
Hi, thanks for the suggestion, it worked perfectly. Should I change anything else?

Reply:
> Hi, thanks for the suggestion, it worked perfectly. Should I change anything else?

Please add a unit-test into `brave/components/embedder_support/user_agent_utils_unittest.cc`

Reply:
Done!